### ISSUE TITLE
[💡SUG] Add KG-Rec as a KG-aware recommender model

### ISSUE THREAD

**Issue by jackmedda** [tail-unica/hopwise#218]
**Describe the model you wish to add and its strengths.**
A new self-supervised rationalization method, called **KGRec**, for knowledge-aware recommender systems. To highlight rationales in the knowledge graph, we design a novel generative task in the form of masking-reconstructing. By masking important knowledge with high rational scores, KGRec is trained to rebuild and highlight useful knowledge connections that serve as rationales. To further rationalize the effect of collaborative interactions on knowledge graph learning, we introduce a contrastive learning task that aligns signals from knowledge and user-item interaction views. To ensure noise-resistant contrasting, potential noisy edges in both graphs judged by the rational scores are masked.

**Describe the type of model**
    Knowledge-aware

**Describe the loss of model**
It seems to be based on three losses, BPR (included in hopwise), MAE, and a Contrastive Loss (CL).

**Describe the Input of model**
It seems it takes only interactions as a triple (user, positive item, negative item), so the learning from KG is not dependent on KG triples input. It follows that the [KnowledgeBasedDataloader](https://github.com/tail-unica/hopwise/blob/main/hopwise/data/dataloader/knowledge_dataloader.py) mode should be always set to `KGDataLoaderState.RS`.

**Provide the original paper or source**
https://arxiv.org/pdf/2307.02759 (KDD, 2023)

**Code availability**
https://github.com/HKUDS/KGRec (PyTorch)

**Implementation Guidelines**
- Work on the branch created for this issue
- The model should be added as a knowledge-aware recommender in the corresponding [folder](https://github.com/tail-unica/hopwise/tree/main/hopwise/model/knowledge_aware_recommender)
- Follow other files to standardize the comments on top of the file, and also provide docstrings for the class and for important functions (not the ones used by hopwise)
- It's important to implement the key methods that hopwise uses by default (other methods can only exist altogether with the key methods), which are `calculate_loss`, `forward`, `predict`, `full_sort_predict` (if possible)
- Use [KGIN](https://github.com/tail-unica/hopwise/blob/main/hopwise/model/knowledge_aware_recommender/kgin.py) (maybe also [KGCN](https://github.com/tail-unica/hopwise/blob/main/hopwise/model/knowledge_aware_recommender/kgcn.py)) as references to help you in the development
- Use [hopwise documentation](http://tail-unica.github.io/hopwise/) to help you in understanding key components
- It's important to commit often to monitor the different updates on the code
- When finished or if reviews on the code is needed, a pull request can be opened and add @jackmedda as reviewer to give comments on specific aspects (don't abuse it, as it automatically launches testing)
- To test out the model, you'll also need to create a "properties" file inside the [model properties folder](https://github.com/tail-unica/hopwise/tree/main/hopwise/properties/model). The file is a YAML file that should be named exactly as the model class (the hyphen can be removed from both the class name and the file). The file should only include specific parameters expected by the model, not other parameters related to training or data setup
- Finally, add a function to test the new model in [test_model_auto](https://github.com/tail-unica/hopwise/blob/main/tests/model/test_model_auto.py) under the `TestKnowledgeRecommender` class.
